code/jvm/src/main/kotlin/org/ionproject/codegarten/controllers/ErrorController.kt

```diff
@@ -0,0 +1,36 @@
+package org.ionproject.codegarten.controllers
+
+import org.ionproject.codegarten.Routes.ERROR_HREF
+import org.ionproject.codegarten.responses.ProblemJson
+import org.ionproject.codegarten.responses.Response
+import org.ionproject.codegarten.responses.toResponseEntity
+import org.springframework.boot.web.servlet.error.ErrorController
+import org.springframework.http.HttpStatus
+import org.springframework.http.ResponseEntity
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.RequestAttribute
+import org.springframework.web.bind.annotation.RestController
+import javax.servlet.RequestDispatcher
+import javax.servlet.http.HttpServletRequest
+
+@RestController
+class ErrorController : ErrorController {

```

Comment:
When is this used? Please add this info as a comment.

Reply:
This is used to replace the default Spring whitelabel error page. This allows us to return the errors as a problem JSON response.